proposals/0002-cxx-attributes.md

```diff
@@ -124,4 +124,67 @@ Below are a few more examples of C++ attributes that we could support:
     }
   }
 ```
+
+## Detailed design
+
+The simplest explanation of this feature is supporting C++11 attribute syntax on
+all shared grammar elements between HLSL and C++. This spec attempts to detail
+some of the grammar and parsing implications and will specify the process by
+which existing attributes will convert to C++ attributes.
+
+### Attribute Parsing
+
+This proposal introduces grammar formulations for parsing attributes matching
+some formulations from C++ `dcl.attr.grammar`. Specifically:
+
+```ebnf
+attribute-specifier-seq:    [ attribute-specifier-seq ], attribute-specifier
+attribute-specifier:        "[" , "[" , attribute-list , "]" , "]"
+attribute-list:             [ attribute ] ||
+                            attribute-list , "," , [attribute] ||
+                            { attribute } ||
+                            attribute-list ,  "," , { attribute }
+attribute:                  attribute-token , [attribute-argument-clause]

```

Comment:
```suggestion
attribute:                  attribute-token , [balanced-token]

```

I think you want `balanced-token` here. Otherwise users have to doube the outer parens

```
[[foo()]] ->  specifier( list ( attribute ( identifier argument-clause (empty) ) )
[[foo((bar))]] -> specifier( list ( attribute ( identifier argument-clause ( balanced-token-seq ( balanced-token (balanced token-seq  (empty ) ) ) ) )
[[foo(bar)]] -> specifier( list ( attribute ( identifier argument-clause ( ??? ) ) ).  ??? is bar, but bar doesn't fit into balanced-token-seq
```